fraction_authoring.adoc

```diff
@@ -178,42 +256,56 @@ public class MyComponents implements Whatever {
 }
 ----
 
-==== `ArchivePreparer`
+==== `DeploymentProcessor`
 
 If your fraction needs an opportunity to alter or otherwise prepare all deployed
-archives, you may implement the `org.wildfly.swarm.spi.api.ArchivePreparer` interface.
+archives, you may implement the `org.wildfly.swarm.spi.api.DeploymentProcessor` interface.
 
 [source,java]
 ----
-@Singleton
-public class MyArchivePreparer implements ArchivePreparer {
+@DeploymentScoped
+public class MyArchivePreparer implements DeploymentProcessor {
 
-  @Inject
-  private MyFraction myFraction;
+    private final Archive archive;
+
+    @Inject
+    private MyFraction myFraction;
+
+    @Inject
+    public HttpSecurityPreparer(Archive archive) {
+        this.archive = archive;
+    }
 
-  public void prepareArchive(Archive<?> archive) {
-    archive.as( WARArchive.class ).setContextRoot( myFraction.getContextRoot() );
-  }
```

Comment:
There should probably be a skeleton of the `process` method, otherwise this isn't really a `DeploymentProcessor`.